build.sh

```diff
@@ -2,17 +2,24 @@
 
 set -e
 
-BUILDTYPE=$1
-
-if [ -z "${BUILDTYPE}" ]
-then
-  BUILDTYPE=release
-fi
+case $1 in
+  plain|debug|debugoptimized|release|minsize)
+    BUILDTYPE=$1
+    shift
+    ;;
+  *)
+    BUILDTYPE=release
+    ;;
+esac
 
 BUILDDIR=build/${BUILDTYPE}
 
-rm -fr ${BUILDDIR}
-meson ${BUILDDIR} --buildtype ${BUILDTYPE} --prefix ${INSTALL_PREFIX:-/usr/local}
+if [ -d ${BUILDDIR} ]
+then
+  meson configure ${BUILDDIR} --buildtype ${BUILDTYPE} --prefix ${INSTALL_PREFIX:-/usr/local} "$@"
+else
+  meson ${BUILDDIR} --buildtype ${BUILDTYPE} --prefix ${INSTALL_PREFIX:-/usr/local} "$@"
+fi
```

Comment:
Surely this would have been better as `if -d... mesoncmd='configure' else '' fi; meson $mesoncmd $otherargs` :P